partiql-parser/src/main/kotlin/org/partiql/parser/internal/PartiQLParserDefault.kt

```diff
@@ -1113,6 +1113,17 @@ internal class PartiQLParserDefault : PartiQLParser {
          *
          */
 
+        /**
+         * Verifies if all of the [args] are
+         * 1. [Expr.SFW] or
+         * 2. [Expr.BagOp] and is a SQL Set op (i.e. [Expr.BagOp.outer] != true)
+         */
+        private fun argsAreSFW(args: List<Expr>): Boolean {
+            return args.all { arg ->
+                arg is Expr.SFW || (arg is Expr.BagOp && arg.outer != true)
+            }
+        }
```

Comment:
New parsing logic to distinguish between SQL set ops and PartiQL bag ops
- SQL set ops
  - require no `OUTER` before set operation and
  - require args are SQL set ops or SFW expression
- PartiQL bag ops
  - use `OUTER` keyword or
  - do not use `OUTER` keyword but one or more arguments is a PartiQL bag op or non-SFW expression